src/pharmpy/model/external/nonmem/update.py

```diff
@@ -1473,7 +1473,13 @@ def update_estimation(control_stream, model):
         # Add $COV
         last_est_rec = control_stream.get_records('ESTIMATION')[-1]
         idx_cov = control_stream.records.index(last_est_rec)
-        covrec = create_record('$COVARIANCE\n')
+        if new_cov == 'SANDWICH':
+            covrec_ = '$COVARIANCE'
+        elif new_cov == 'CPG':
+            covrec_ = '$COVARIANCE MATRIX=S'
+        elif new_cov == 'OFIM':
+            covrec_ = '$COVARIANCE MATRIX=R'
+        covrec = create_record(f'{covrec_}\n')
```

Comment:
Could be good to have some code that tests this new logic, e.g. in this test: https://github.com/pharmpy/pharmpy/blob/60ec348d5e52f7f0a9f29587d99999dc0023a489/tests/nonmem/test_nonmem_model.py#L682

Reply:
Added: https://github.com/pharmpy/pharmpy/blob/c4c5dae3950ff7fba6e4cbac2d4d6895f41afdc8/tests/nonmem/test_nonmem_model.py#L658-L659